src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -162,6 +162,18 @@ bool Parser::parse_and_visit_statement(Parse_Visitor_Base &v,
       Expression *ast =
           this->parse_expression(v, Precedence{.in_operator = true});
       this->visit_expression(ast, v, Variable_Context::rhs);
+
+      // let ({x} = y);  // Ambiguous if 'let' is a function
+      if (ast->kind() == Expression_Kind::Call) {
+        Expression::Call *call = expression_cast<Expression::Call *>(ast);
+        if (call->child_count() == 2 &&
```

Comment:
Must fix: You are missing a check for `call->child_0()`. This code reports a diagnostic for `let.prop({x}=y);` but that code should not have a diagnostic.